experiments/explanation/summarization.py

```diff
@@ -69,36 +68,41 @@ def extract_text(beginning_page, last_page):
 
 
 
-def get_explanation(model, text_input, guide):
-    """Get an explanation of provided text from openai"""
+# creates an explanation of the text for the user
+def get_explanation(model, text_input):
+    prompt = ChatPromptTemplate.from_messages([
+        SystemMessagePromptTemplate.from_template("""You are a helpful chatbot that is proficient in explaining text that is passed to it"""),
+        MessagesPlaceholder(variable_name="history"),
+        HumanMessagePromptTemplate.from_template("{input}")
+    ])
     response = []
-
-    user_content = (f"""Process the following text based on the following guidelines:
-                        Provided Guidelines: {guide}
-                        Provided Text: {text_input}"""
-                    if guide else f"""You will explain the following text: {text_input}""")
-
-    bit_response = openai.ChatCompletion.create(model=model,
-                                                messages=[
-                                                       {"role": "user", "content": f"{user_content}"}
-                                                ],
-                                                stream=True)
-    for chunk in bit_response:
-        if not chunk.choices[0].delta:
-            break  # End of the stream; break the loop
-
-        response.append(chunk.choices[0].delta.content)  # save the event response
+    conversation = ConversationChain(memory=st.session_state.memory, 
+                                    prompt=prompt,
+                                    verbose= True, 
+                                    llm=ChatOpenAI(temperature=0.6, model= model, streaming=True, callbacks=[MyCustomSyncHandler()]),)
+    bit_response = conversation.predict(input=text_input)
+
+    for chunk in bit_response: # This for loop is only needed for token count
```

Comment:
Awesome, this prints the stream right?

Reply:
Yes! Each time there is a new token, the callback function prints it.